lib/queue.rb

```diff
@@ -1,31 +1,59 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = @back = -1
+    @data = ''
+    # - 1 gets put into @back and the value of @back is put into @front 
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
-  end
+    # ## if the queue is empty
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+      @store[@front] = element
+    elsif @front == @back
+      # if the queue if full
+      raise ArgumentError.new('The queue is full') 
+    else
+      @store[@back] = element
+      @back = (@back + 1) % @store.length
+    end 
+  end 
+
+  
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1
+      raise ArgumentError, "Queue Empty"
+    else
+      value = @store[@front]
+      @store[@front] = -nil
+      @front = (@front +  1) % @max_size
+      if @front == @back 
+        @front = @back = -1
+      end
+      return value
+    end
+ 
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return @store.length
```

Comment:
This only gives the size of the internal array size, not the number of elements in the Queue.